hexrd/material.py

```diff
@@ -707,15 +707,17 @@ def U(self):
     @U.setter
     def U(self, Uarr):
         Uarr = numpy.array(Uarr)
-        if self.unitcell.U.shape == Uarr.shape:
-            if not numpy.allclose(self.unitcell.U, Uarr):
-                self.unitcell.U = Uarr
-                self.update_structure_factor()
-            else:
-                return
-        else:
-            self.unitcell.U = Uarr
-            self.update_structure_factor()
+        self._U = Uarr
+        # self.unitcell.U = Uarr
+        # if self.unitcell.U.shape == Uarr.shape:
+        #     if not numpy.allclose(self.unitcell.U, Uarr):
+        #         self.unitcell.U = Uarr
+        #         self.update_structure_factor()
+        #     else:
+        #         return
+        # else:
+        #     self.unitcell.U = Uarr
+        #     self.update_structure_factor()
```

Comment:
So we are removing these updates since we should only modify the data with `_set_atomdata` now?

Reply:
@saransh13 -- can we resolve this today?

Reply:
This is resolved.